File: kube-guard-frontend/src/api/namespaces.ts
// src/api/namespaces.ts
import { get, post } from '@/api/http'

export type NamespaceItem = {
  name: string
  labels: Record<string, string>
  phase?: string | null
  createdAt?: string | null
  system: boolean
  managed: boolean
  managedByTenantId?: string | null
}

export function fetchNamespaceOptions(): Promise<{ items: string[] }> {
  return get('/api/namespaces/options')
}

export function fetchNamespaces(params?: { keyword?: string }): Promise<{ items: NamespaceItem[] }> {
  return get('/api/namespaces', { params })
}

export function createNamespace(body: {
  name: string
  labels?: Record<string, string>
  managed?: boolean
  managedByTenantId?: string
}): Promise<{ item: NamespaceItem }> {
  return post('/api/namespaces', body)
}

export async function patchNamespaceLabels(
  name: string,
  labels: Record<string, string>
): Promise<{ item: NamespaceItem }> {
  // 你没写 patch wrapper，这里直接用 get/post 思路写一个最简单的：
  const { http } = await import('@/api/http')
  const resp = await http.patch(`/api/namespaces/${encodeURIComponent(name)}/labels`, { labels })
  return resp.data as { item: NamespaceItem }
}

export async function deleteNamespace(name: string, purgeRegistry = true): Promise<{ ok: boolean }> {
  const { http } = await import('@/api/http')
  const resp = await http.delete(`/api/namespaces/${encodeURIComponent(name)}`, {
    params: { purge_registry: purgeRegistry },
  })
  return resp.data as { ok: boolean }
}

File: backend/kube-devops-backend/fastApiProject/routers/namespaces.py
# routers/namespaces.py
from __future__ import annotations

from typing import Dict, Optional
from fastapi import APIRouter, Query
from pydantic import BaseModel

from services.namespaces import service as svc

router = APIRouter(prefix="/api/namespaces", tags=["Namespaces"])

class NamespaceCreateIn(BaseModel):
    name: str
    labels: Dict[str, str] = {}
    managed: bool = True
    managedByTenantId: Optional[str] = None

class NamespaceLabelsPatchIn(BaseModel):
    labels: Dict[str, str] = {}

@router.get("")
def get_namespaces(keyword: Optional[str] = Query(default=None)):
    return {"items": svc.list_namespaces(keyword=keyword)}

@router.get("/options")
def get_namespace_options():
    return {"items": svc.list_namespace_options()}

@router.post("")
def post_namespace(body: NamespaceCreateIn):
    item = svc.create_namespace(
        name=body.name,
        labels=body.labels,
        managed=body.managed,
        managed_by_tenant_id=body.managedByTenantId,
    )
    return {"item": item}

@router.patch("/{name}/labels")
def patch_ns_labels(name: str, body: NamespaceLabelsPatchIn):
    item = svc.patch_namespace_labels(name=name, labels=body.labels)
    return {"item": item}

@router.delete("/{name}")
def delete_ns(name: str, purge_registry: bool = True):
    return svc.delete_namespace(name=name, purge_registry=purge_registry)
